OpenRA.Mods.Common/MapGenerator/TilingPath.cs

```diff
@@ -803,7 +803,10 @@ void UpdateFrom(CVec from, int fromTypeId, bool isForStart)
 				}
 
 				Debug.Assert(candidates.Count >= 1, "TraceBack didn't find an original route");
-				var chosenSegment = candidates[random.Next(candidates.Count)];
+				var weights = candidates
+					.Select(c => c.MultiBrush.Weight)
+					.ToArray();
```

Comment:
```suggestion
				var weights = candidates.ConvertAll(c => c.MultiBrush.Weight);
```

We generally want to use lists for such dynamic allocations. It's less memory cost than an array (you need to make memory for building the array, and then also allocate an array after you've done collecting members )